core/metrics-core-service/src/main/java/org/hawkular/metrics/core/dropwizard/HawkularObjectNameFactory.java

```diff
@@ -51,9 +50,15 @@ private boolean needsQuote(String propertyValue){
     public ObjectName createName(String type, String domain, String name) {
         try {
             Hashtable<String, String> table = new Hashtable<>();
-            table.putAll(metaDataMap.get(name).getTags());
-            table.replaceAll((k,v) -> needsQuote(v) ? ObjectName.quote(v): v );
-            return new ObjectName(domain, table);
+            MetaData metaData = metricRegistry.getMetaData(name);
+            if (metaData == null) {
+                LOGGER.warnf("No meta data found for %s", name);
+                return new ObjectName(domain, type, name);
+            } else {
+                table.putAll(metricRegistry.getMetaData(name).getTags());
```

Comment:
The ObjectName constructor unfortunately takes a Hashtable and not a Map; so, we are forced into copying the tags into ```table```.